libcaf_core/caf/detail/type_list.hpp

```diff
@@ -228,6 +228,35 @@ struct tl_at<type_list<E...>, N> {
 template <class List, size_t N>
 using tl_at_t = typename tl_at<List, N>::type;
 
+// list remove(list, value)
+
+template <class Input, class Value, class Output>
+struct tl_remove_impl;
+
+template <class Value, class... Out>
+struct tl_remove_impl<empty_type_list, Value, type_list<Out...>> {
+  using type = type_list<Out...>;
+};
+
+template <class T, class... Ts, class Value, class... Out>
+struct tl_remove_impl<type_list<T, Ts...>, Value, type_list<Out...>> {
+  using type = std::conditional_t<
+    std::is_same_v<T, Value>,
+    typename tl_remove_impl<type_list<Ts...>, Value, type_list<Out...>>::type,
+    typename tl_remove_impl<type_list<Ts...>, Value,
+                            type_list<Out..., T>>::type>;
+};
```

Comment:
If I got this right, the implementation will reverse the items in the type list:
```
tl_remove<type_list<int, string, double, float>, float> => type_list<double, string, int>
```
I'd rather have the ordering preserved (with the appropriate item removed). This might not be important to the compiler, but I'd rather not confuse the users who read compiler logs more than they already are. 
```suggestion
template <class T, class... Ts, class Value, class... Out>
struct tl_remove_impl<type_list<T, Ts...>, Value, type_list<Out...>> {
  using type = std::conditional_t<
    std::is_same_v<T, Value>,
    typename tl_remove_impl<type_list<Ts...>, Value, type_list<Out...>>::type,
    typename tl_remove_impl<type_list<Ts...>, Value,
                            type_list<T, Out...>>::type>;
};
```

Reply:
Sorry, I made a mistake here, it's too late for recursive templates.